src/m3ninx/index/segment/builder/builder.go

```diff
@@ -109,13 +148,11 @@ func (b *builder) Insert(d doc.Document) ([]byte, error) {
 func (b *builder) InsertBatch(batch index.Batch) error {
 	// NB(r): This is all kept in a single method to make the
 	// insertion path fast.
+	var wg sync.WaitGroup
```

Comment:
Hm, maybe we should just use one on the builder itself?  Then we can reuse it each time.  

If you take a reference to the `wg` I believe it will allocate it on the heap (since escape analysis doesn't know the lifetime of the functions using this pointer, they could keep a ref to it and call it minutes later, etc).